aten/src/ATen/native/MathBitsFallback.h

```diff
@@ -72,55 +87,79 @@ struct MathOpFallback {
       return;
     }
 
-    // Mutable inputs to be tracked separately
-    std::vector<Tensor> mutable_inputs;
+    /*
+    Different possible cases:
+    1. Functions with no tensorlist inputs
+      a. no mutable args
+      b. one or more mutable args
+         b.1. shared memory between mutable and non-mutable args (--handled)
+         b.2. shared memory between two or more mutable args (incorrect result but this is bad
+              and users shouldn't do it anyway)
+         b.3. shared memory between two or more non-mutable args (works fine since we never modify the memory)
+         b.4. no shared memory between args (-- works fine)
+    2. Functions with tensorlist inputs
+      a. no mutable args (-- works fine)
+      b. Mutable tensor arg(s) but non-mutable tensorlist arg
+        b.1. All the possible cases listed in 1.b (-- works fine)
+        b.2. shared memory between a tensor arg and a tensorlist arg ( -- Not currently supported)
+      c. Mutable tensorlist arg(s) ( -- Not currently supported)
+        c.1. shared memory between a mutable and non-mutable tensorlist
+        c.2. shared memory between two or more mutable tensorlist args
+        c.3. shared memory between two or more non-mutable tensorlist args
+        c.4. shared memory between two or more non-mutable tensor args
+        c.5. shared memory between a tensor and tensorlist arg
+        ...
+        c.n. no shared memory between args
+    */
 
+    // updates for non-mutable inputs
     for (const auto i : c10::irange(num_arguments)) {
       auto& ivalue = (*stack)[stack_start + i];
       if (!(ivalue.isTensor() || ivalue.isTensorList())) {
         continue;
       }
       const auto& argument = arguments[i];
-      bool mut_arg = false;
       if (argument.alias_info()) {
         // Was already tested by is_write loop above
         TORCH_INTERNAL_ASSERT_DEBUG_ONLY(argument.alias_info()->isWrite());
-        mut_arg = true;
+        continue;
```

Comment:
continue'ing here looks sus. If the argument is mutable AND conjugate bit is set, passing it on directly seems wrong because the kernel is going to ignore the conjugate bit and stomple all over it.

Reply:
no because we only wanna handle the non-mutable args in this loop and not modify the memory until we verify either the non-mutable args don't share storage with mutable args (with conj bit set to `True`) else clone them. 

Then we loop through the mutable args with conj bit set to `True`, and in-place conjugate them.